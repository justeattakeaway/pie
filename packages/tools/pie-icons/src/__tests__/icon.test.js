/* eslint-env jest */
import Icon, { normaliseClassname } from '../icon';

const icon1 = new Icon(
    'test',
    '<line x1="23" y1="1" x2="1" y2="23" /><line x1="1" y1="1" x2="23" y2="23" />',
    '',
    '',
    ['hello', 'world', 'foo', 'bar'],
);

const icon2 = new Icon(
    'test',
    '<line x1="23" y1="1" x2="1" y2="23" /><line x1="1" y1="1" x2="23" y2="23" />',
);

test('constructs icon object correctly', () => {
    expect(icon1).toMatchSnapshot();
    expect(icon2).toMatchSnapshot();
});

describe('toSvg()', () => {
    test('returns correct string when run with no additional attrs', () => {
        expect(icon1.toSvg()).toMatchSnapshot();
    });
    test('appends color and stroke-width attributes when passed through', () => {
        expect(icon1.toSvg({ color: 'red', 'stroke-width': 1 })).toMatchSnapshot();
    });
    test('appends classnames and attributes when passed through as attrs', () => {
        expect(icon1.toSvg({ class: 'foo bar', color: 'red' })).toMatchSnapshot();
    });

    test('adds React Native default attrs when platform = "reactNative"', () => {
        expect(icon1.toSvg({}, 'reactNative')).toMatchSnapshot();
    });
});

test('toString() returns correct string', () => {
    expect(icon1.toString()).toMatchSnapshot();
});

describe('normaliseClassname()', () => {
    test('returns camelCased String when capitalised string is passed in', () => {
        const classname = 'Alert';
        expect(normaliseClassname(classname)).toBe('alert');
    });

    test('returns camelCased String when dashed string is passed in', () => {
        const classname = 'alert-filled-large';
        expect(normaliseClassname(classname)).toBe('alertFilledLarge');
    });

    test('returns camelCased String when dashed capitalised string is passed in', () => {
        const classname = 'Alert-filled-large';
        expect(normaliseClassname(classname)).toBe('alertFilledLarge');
    });
});
