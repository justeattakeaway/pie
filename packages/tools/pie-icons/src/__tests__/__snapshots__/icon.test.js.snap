// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`constructs icon object correctly 1`] = `
Icon {
  "attrs": {
    "fill": "currentColor",
    "focusable": false,
    "role": "presentation",
    "xmlns": "http://www.w3.org/2000/svg",
  },
  "contents": "<line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\" /><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\" />",
  "name": "test",
}
`;

exports[`constructs icon object correctly 2`] = `
Icon {
  "attrs": {
    "fill": "currentColor",
    "focusable": false,
    "role": "presentation",
    "xmlns": "http://www.w3.org/2000/svg",
  },
  "contents": "<line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\" /><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\" />",
  "name": "test",
}
`;

exports[`toString() returns correct string 1`] = `"[object Object]"`;

exports[`toSvg() > adds React Native default attrs when platform = "reactNative" 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"presentation\\" focusable=\\"false\\" fill=\\"orange\\" width=\\"50\\" height=\\"50\\"><line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\" /><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\" /></svg>"`;

exports[`toSvg() > appends classnames and attributes when passed through as attrs 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"presentation\\" focusable=\\"false\\" fill=\\"red\\" class=\\"c-pieIcon c-pieIcon--test foo bar\\"><line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\" /><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\" /></svg>"`;

exports[`toSvg() > appends fill and stroke-width attributes when passed through 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"presentation\\" focusable=\\"false\\" fill=\\"red\\" stroke-width=\\"1\\" class=\\"c-pieIcon c-pieIcon--test\\"><line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\" /><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\" /></svg>"`;

exports[`toSvg() > returns correct string when run with no additional attrs 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"presentation\\" focusable=\\"false\\" fill=\\"currentColor\\" class=\\"c-pieIcon c-pieIcon--test\\"><line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\" /><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\" /></svg>"`;
